src/main/java/envoy/client/net/Client.java

```diff
@@ -124,22 +117,23 @@ public void performHandshake(LoginCredentials credentials,
 	public void initReceiver(LocalDB localDB, CacheMap cacheMap) throws IOException {
 		checkOnline();
 
+		// Remove all processors as they are only used during the handshake
+		receiver.removeAllProcessors();
+
 		// Process incoming messages
-		final ReceivedMessageProcessor			receivedMessageProcessor			= new ReceivedMessageProcessor();
-		final ReceivedGroupMessageProcessor		receivedGroupMessageProcessor		= new ReceivedGroupMessageProcessor();
-		final MessageStatusChangeProcessor		messageStatusChangeProcessor		= new MessageStatusChangeProcessor();
-		final GroupMessageStatusChangeProcessor	groupMessageStatusChangeProcessor	= new GroupMessageStatusChangeProcessor();
+		final var	receivedMessageProcessor			= new ReceivedMessageProcessor();
+		final var	receivedGroupMessageProcessor		= new ReceivedGroupMessageProcessor();
+		final var	messageStatusChangeProcessor		= new MessageStatusChangeProcessor();
+		final var	groupMessageStatusChangeProcessor	= new GroupMessageStatusChangeProcessor();
 
 		receiver.registerProcessor(GroupMessage.class, receivedGroupMessageProcessor);
 		receiver.registerProcessor(Message.class, receivedMessageProcessor);
 		receiver.registerProcessor(MessageStatusChange.class, messageStatusChangeProcessor);
 		receiver.registerProcessor(GroupMessageStatusChange.class, groupMessageStatusChangeProcessor);
 
-		// Relay cached unread messages and unread groupMessages
+		// Relay caches
```

Comment:
The current comment is just so non-descriptive that even I (one of Envoy's programmers) cannot understand it. 
```suggestion
		// Relay stored unread (group-) messages
```